### PR TITLE
feat(cards): switch and fallback card options

### PR DESCRIPTION
**Pull request by fx**
## Summary

Second PR of `docs/changes/0022-switch-input-helpers-to-spec.md`: the switch and fallback card's option surface — `confirm` gating, `deviceClassIcon`, and `stateLabels`. `ButtonCard.tsx` becomes a `ButtonCard/` folder per the component-folder convention, with its icon resolution and last-changed formatting extracted as unit-testable modules.

## Confirmation gates classify by effect, not by service name

The common option contract is explicit that a gate enumerating only `domain.*` services is a bypass: the generic `homeassistant.toggle` / `turn_on` / `turn_off` aliases reach the same entity and must clear the same gate. `confirmableService` is therefore a pure classifier over the *resolved* action rather than a service-name allow-list, and it sits between action resolution and dispatch — the gate needs the resolved action because it classifies by effect, while the dispatch guard downstream keys on the final payload.

The classifier's decline paths are tested as deliberately as its accept paths, since a gate that returns true for everything passes every "did it gate?" test anyone writes. Every branch is exercised on both sides, with real cases behind each refusal: a non-confirmable service (`switch.reload`), a `data.entity_id` naming a different entity, a foreign domain (`light.turn_on` against a switch entity), the non-toggling action types, and an unmapped domain (`siren.turn_on`) — alongside the generic aliases the contract requires.

## Scope

`ButtonCard`'s own toggle is deliberately **not** migrated onto the guarded dispatch path here. That path is currently reachable only from the shell's gesture controller and is private to it, so there is nothing for a card's own control to call yet — see #230. The switch option doc records the gap, and this change's final PR verifies the migration rather than repeating it.

## Testing

- [x] `npm test` — 2188 passed, 2 skipped
- [x] `npm run lint`
- [x] `npm run typecheck`
- [x] `npm run test:coverage` — patch clean across all changed source files, verified against `BRDA` branch entries rather than the percentage summary
- [x] `npm run build-storybook` and `npm run build:ha:prod`

### CODE REVIEW

File: src/hooks/useCardActions.ts [hunk 1]
```diff
@@ -55,6 +56,62 @@ export interface UseCardActionsOptions {
...
+  if (!entityId) return null
+  if (action === 'toggle') return 'toggle'
+  if (typeof action !== 'object' || action.action !== 'call-service') return null
+
+  // The effective target: an explicit `data.entity_id` wins at dispatch, so an
+  // action aimed at another entity is not this card's toggle to gate.
+  const explicit = action.data?.entity_id
+  const target = typeof explicit === 'string' ? explicit : entityId
+  if (target !== entityId) return null
+
```
[the remark this answers is not in the document]
Comment (fx) [reply src/hooks/useCardActions.ts]: Fixed in dc7ba5a, and chasing it properly found two more shapes with the same hole — `buildServiceData` spreads the whole payload, so `entity_id: 'all'` reached this entity and was waved through, and an entity buried in a multi-entity list is just as consequential as one named alone. Targets now resolve the way dispatch resolves them:

| payload | gated |
| --- | --- |
| absent | yes — the card's entity is the implicit target |
| `'switch.well_pump'` | yes |
| `'other.entity'` | no |
| `['switch.well_pump']` | **yes** — the bypass you reported |
| `['other', 'switch.well_pump']` | **yes** — member of a larger set |
| `['other', 'light.desk']` | no |
| `[]` | no — reaches nothing |
| `'all'` / `['all']` | **yes** |
| `'none'` | no |
| unresolvable (number, `null`, object) | **yes** — fails toward the visible error rather than the silent dispatch |

Tests enumerate the shapes, plus a dispatch-boundary case proving an array-targeted action reaches `hass.callService` only after the dialog is confirmed. Probed by reinstating the bypass, which fails 4 tests.

Worth recording why our own verification missed this: the classifier had complete branch coverage, all 18 outcomes non-zero, and it could not have caught it. `typeof explicit === 'string'` was 5/25 — both sides taken — and the false side meant "not a string, so assume it misses us", which is exactly where the bug lived. Branch coverage proves the branches you wrote are tested; it says nothing about the branches you failed to write. The target-shape enumeration is now a MUST in the switch doc's `confirm` section so the next implementer inherits it rather than rediscovering it.

File: docs/specs/entity-cards/index.md [hunk 2]
```diff
@@ -148,14 +148,23 @@ See [card reference — Input helpers](./card-reference.md#input-helper-cards),
...
+- `ButtonCard` MUST show `ERROR` / `UNAVAILABLE` states and MUST expose the configuration modal — the [switch option surface](./options/switch.md) (`confirm`, `deviceClassIcon`, `stateLabels`, `showLastChanged`) plus the universal options.
+- Configuration MUST resolve through the card that renders, not the raw entity domain: an unmapped domain resolves to the fallback card's option set, so the options are editable wherever the fallback is what the user sees.
+- `confirm` MUST gate every route that toggles the card's own entity — `default`, an explicit `toggle` on any gesture, and a `call-service` naming `toggle`/`turn_on`/`turn_off` in the entity's own domain or the generic `homeassistant` domain — and MUST gate nothing else. The gate MUST be applied after action resolution, in the shell, so a re-routed action cannot reach the entity around it.
+- Every option MUST be safe on an arbitrary domain: no crash, no `device_class` lookup, and states other than `on`/`off` rendered exactly as reported.
```
[the remark this answers is not in the document]
Comment (fx) [reply docs/specs/entity-cards/index.md]: Fixed in dc7ba5a — the index's "Button and fallback card" section is down to four high-level bullets and a pointer to `options/switch.md` for the option surface, icon precedence, gate scope and fallback-safety rules. The confirm-scope scenario came out of the index too, since it duplicated the option doc's own.

File: docs/specs/entity-cards/options/switch.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
 # Card Options — Switch
 
-Part of the [entity-cards spec](../index.md); builds on the [common contract](./common.md) (universal options are not repeated here). **Status: specified, not yet implemented.**
+Part of the [entity-cards spec](../index.md); builds on the [common contract](./common.md) (universal options are not repeated here). **Status: implemented** by change [0022](../../../changes/0022-switch-input-helpers-to-spec.md), except the transition-or-timeout dispatch guard noted under Primary action.
 
 This document covers the switch card: the card registered for the `switch` domain and, unchanged, the **generic fallback card** rendered for any entity domain without a registry entry (see [entity-cards — Button and fallback card](../index.md#button-and-fallback-card), `src/components/ButtonCard.tsx`). Because the same card serves arbitrary domains, every option below MUST be safe — no crash, no misleading UI — when the entity is not a `switch.*` entity.
```
[the remark this answers is not in the document]
Comment (fx) [reply docs/specs/entity-cards/options/switch.md]: Fixed in dc7ba5a. Grepping the tree found it in more places than the two lines: `options/switch.md`, `options/scene.md` (×2), `card-reference.md` (with a stale line number alongside it), and the index's file list. All now point at `src/components/ButtonCard/`.

One deliberate exception: `docs/changes/0010-design-tokens-and-anatomy.md` still says `ButtonCard.tsx`. That is a merged change document describing the tree as it stood at the time, and updating it would make it inaccurate rather than current.
